src/components/Base/BaseToggleGroup.vue

```diff
@@ -0,0 +1,44 @@
+<template>
+  <ToggleGroupRoot
+    class="ToggleGroup"
+    :type
+    :modelValue
+    :disabled
+    @update:modelValue="
+      (value) => {
+        if (value != undefined) {
+          $emit('update:modelValue', value);
+        }
+      }
+    "
+  >
+    <slot />
+  </ToggleGroupRoot>
+</template>
+
+<script setup lang="ts">
+import { ToggleGroupRoot } from "radix-vue";
+
+defineProps<{
+  type: "single" | "multiple";
+  modelValue: string | string[];
+  disabled?: boolean;
+}>();
```

Comment:
modelValueの値に特定のリテラル文字列しか含められないよう、genericsを使いたくなるかも？
実はVueの引数もgenericsが使えたりします！
https://ja.vuejs.org/api/sfc-script-setup#generics

こうすると良い感じになる･･････？（自信なし）

```suggestion
<script setup lang="ts" generic="T extends string">
import { ToggleGroupRoot } from "radix-vue";

defineProps<{
  type: "single" | "multiple";
  modelValue: T | T[];
  disabled?: boolean;
}>();
```

なんか難しそうだったら諦める感じで全然大丈夫だと思います！！

Reply:
試してみたけど、ToggleGroupRootがgenericsに対応してなさそうでした！